source/community.applications/usr/local/emhttp/plugins/community.applications/include/previous_apps_helpers.php

```
@@ -487,6 +487,35 @@ private static function collectInstalledLanguagePacks($templates, $isActionCentr
+	 * single sweep so the sidebar can enable/disable the submenu items
+	 * without rendering full result pages.
+	 *
+	 * Mirrors the same context previous_apps() builds — feed array, moderation
+	 * overrides, docker info / update status — so the counts agree byte-for-byte
+	 * with what each submenu would actually render.
+	 *
+	 * @return array{instDocker:int,instPlugins:int,prevDocker:int,prevPlugins:int}
+	 */
+	public static function installedAndPreviousCounts(): array {
+		$file = &$GLOBALS['templates'];
+		$extraBlacklist = readJsonFile(CA_PATHS['extraBlacklist']) ?: [];
```

PHPStan (2.1.54)

[error] Constant CA_PATHS not found.
Learn more at https://phpstan.org/user-guide/discovering-symbols

(constant.notFound)